packages/appeals-service-api/src/repositories/sql/appeals-repository.js

```diff
@@ -0,0 +1,76 @@
+const { createPrismaClient } = require('../../db/db-client');
+const { Prisma } = require('@prisma/client');
+const logger = require('../../lib/logger');
+const ApiError = require('../../errors/apiError');
+
+/**
+ * @typedef { import("@prisma/client").Appeal } Appeal
+ * @typedef { import("@prisma/client").Prisma.AppealCreateInput } AppealCreateInput
+ */
+
+class AppealsRepository {
+	dbClient;
+
+	constructor() {
+		this.dbClient = createPrismaClient();
+	}
+
+	/**
+	 * Create a new Appeal
+	 *
+	 * @param {AppealCreateInput} appeal
+	 * @returns {Promise<Appeal>}
+	 */
+	async createAppeal(appeal) {
+		try {
+			return await this.dbClient.appeal.create({
+				data: appeal
+			});
+		} catch (err) {
+			logger.error(err);
+			if (err instanceof Prisma.PrismaClientKnownRequestError) {
```

Comment:
This check isn't enough to know that it's a duplicate (this was perhaps copied I missed it elsewhere).

See: https://www.prisma.io/docs/orm/reference/error-reference#prismaclientknownrequesterror - we can check error codes instead.

Reply:
interesting 🤔 I actually inherited this code by copy and pasting the user repo so I'll have a look at the link and fix it there too